backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java

```diff
@@ -71,6 +73,41 @@ public UsersController(WebClientService webClientService, @Value("${config.vanit
 
             searchResults = ResponseEntity.status(searchResults.getStatusCode()).body(filteredUsers);
         }
+        
+        //Filter based on selected client & roles
```

Comment:
I know this method was already too long before you added this, but it's getting pretty long. Considering moving this into something like:
```java
// Filter based on selected client & roles
if (clientId.isPresent()){
    users = filterUsersByRole(clientId, selectedRoles, users);
    searchResults = ResponseEntity.status(searchResults.getStatusCode()).body(users);
}
// ...

private List<Object> filterUsersByRole(Optional<String> clientId, Optional<String[]> selectedRoles, List<Object> users)
```